enhancements/dns/externaldns-operator.md

```diff
@@ -0,0 +1,500 @@
+- [Implementation History](#implementation-history)
+- [Drawbacks](#drawbacks)
+- [Alternatives](#alternatives)
+- [Infrastructure Needed](#infrastructure-needed)
+<!-- /toc -->
+
+## Release Signoff Checklist
+
+- [X] Enhancement is `implementable`
+- [ ] Design details are appropriately documented from clear requirements
+- [ ] Test plan is defined
+- [ ] Graduation criteria for dev preview, tech preview, GA
+- [ ] User-facing documentation is created in [openshift-docs](https://github.com/openshift/openshift-docs/)
+
+## Summary
+
+This enhancement proposal is for adding an operator to manage
+[ExternalDNS](https://github.com/kubernetes-sigs/external-dns). ExternalDNS has been chosen for managing external DNS
+requirements of OpenShift clusters. Initially, the operator will focus on managing external DNS records for OpenShift
+[Routes](https://github.com/openshift/api/blob/master/route/v1/types.go), in addition to Kubernetes Services.
+Additionally, a new ExternalDNS Operator CRD will be used to initially expose limited ExternalDNS configuration capabilities.
+
+Existing operators will be used to guide the design of the ExternalDNS Operator. Tooling will be
+included to make it easy to build, run, test, etc. the operator for OpenShift, and for vanilla Kubernetes.
+
+## Motivation
+
+ExternalDNS is a controller that manages external DNS records. Our primary goal is to integrate ExternalDNS as a supported
+OpenShift component. As with other platform components, ExternalDNS requires an operator to provide lifecycle management of
+the component.
+
+Secondarily, the benefits of operators in OpenShift are well established,
+the [operator pattern](https://kubernetes.io/docs/concepts/extend-kubernetes/operator/) is increasingly accepted in the
+upstream Kubernetes community, and the ExternalDNS community would also theoretically benefit from the introduction
+of an ExternalDNS operator. If such an operator were compatible with vanilla Kubernetes (in addition to OpenShift),
+it could be utilized by other downstream ExternalDNS users to simplify the deployment and lifecycle management of ExternalDNS.
+
+This enhancement proposal aims to bring the benefits of operators to ExternalDNS. Ideally, the upstream ExternalDNS community
+would ultimately participate in the design and construction of the ExternalDNS Operator.
+
+### Goals
+
+* Explore the use of an operator for managing ExternalDNS.
+* Create patterns, libraries, and tooling so that ExternalDNS Operator is of high quality, consistent in its API
+surface (common fields on relevant ExternalDNS CRDs, consistent labeling of created resources, etc.), yet is easy to build.
+* Build an operator that is suitable for production use of ExternalDNS.
+* Focus on supporting the OpenShift Route resource as a source for ExternalDNS.
+* Additionally support the Kubernetes [service source](https://github.com/kubernetes-sigs/external-dns/blob/master/source/service.go) (namely LoadBalancer services)
```

Comment:
Aren't services of type LoadBalancer already supported by ExternalDNS? [FAQ](https://github.com/kubernetes-sigs/external-dns/blob/master/docs/faq.md#which-kubernetes-objects-are-supported)

Reply:
Right, the initial ExternalDNS Operator will only support a subset of the source resources supported by ExternalDNS.